File: online_judge/lintcode/count_and_say.java
public class Solution {
    /**
     * @param n the nth
     * @return the nth sequence
     */
    public String countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        String str = countAndSay(n-1) + "*";
        char[] carr = str.toCharArray();

        int count = 1;
        String s = "";
        for (int i = 0; i < carr.length-1; ++i) {
            if (carr[i] == carr[i+1]) {
                ++count;
            } else {
                s = s + count + carr[i];
                count = 1;
            }
        }
        return s;
    }
}

File: online_judge/leetcode/count-and-say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        res = '1'
        for _ in range(1, n):
            j = 0
            new_res = ''
            while j < len(res):
                count = 0
                val = int(res[j])
                while j < len(res) and val == int(res[j]):
                    count += 1
                    j += 1
                new_res += (str(count) + str(val))
            res = new_res
        return res
